components/script/dom/htmlformelement.rs

```diff
@@ -559,8 +582,28 @@ impl HTMLFormElement {
                     HTMLElementTypeId::HTMLInputElement => {
                         let input = child.downcast::<HTMLInputElement>().unwrap();
                         // Step 3.2-3.7
-                        if let Some(datum) = input.form_datum(submitter) {
-                            data_set.push(datum);
```

Comment:
I'm not sure if this is correct -- [form_datum](https://github.com/izgzhen/servo/blob/af7c0c0eb3095e91acf374e0011a18b5e4fdf4ba/components/script/dom/htmlinputelement.rs#L651) has more involved logic. I think we should just add the File case to that function